src/Api/Models/Request/SendRequestModel.cs

```diff
@@ -71,7 +71,8 @@ public Send ToSend(Send existingSend, ISendService sendService)
                     existingSend.Data = JsonSerializer.Serialize(fileData, JsonHelpers.IgnoreWritingNull);
                     break;
                 case SendType.Text:
-                    existingSend.Data = JsonSerializer.Serialize(ToSendData(), JsonHelpers.IgnoreWritingNull);
+                    var sendData = ToSendData();
+                    existingSend.Data = JsonSerializer.Serialize(sendData, sendData.GetType(), JsonHelpers.IgnoreWritingNull);
```

Comment:
Won't this always be typeof `SendTextData`? I believe we can hardcode that.

Reply:
If we are going to hardcode `SendTextData` I would want to do it on the return type of `ToSendData()`. I wasn't sure if there was a decision made to return the base class from `ToSendData()` because in the future we might have wanted it to return other types and the other way I had it would just support that. But I do like hard coding the return type and we can go down that road later.